src/io/GeantGeometryMap.cc

```diff
@@ -0,0 +1,110 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantGeometryMap.cc
+//---------------------------------------------------------------------------//
+#include "GeantGeometryMap.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct/destruct with defaults
+ */
+GeantGeometryMap::GeantGeometryMap()  = default;
+GeantGeometryMap::~GeantGeometryMap() = default;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the mat_id for a given vol_id
+ */
+const GeantGeometryMap::mat_id
+GeantGeometryMap::get_matid(vol_id& volume_id) const
```

Comment:
I think I may have confused you about when to have values, references, pointers, and when each of those should be const. (In C++ it is rather confusing.)

So let me try to summarize how to use each of those in function arguments. Remember that inside a function, arguments act like local variables.
- Always values whenever the argument type is small (`sizeof(arg) <= sizeof(void*)`). Using values instead of pointers/references allows the compiler to optimize better. If the argument is nontrivial but you need to make a local copy anyway, it's OK to make the function argument a value (and use `std::move` internally as needed, but this is a more complicated topic).
- Generally don't use `const` values (e.g. `const int`), because the decision of whether to modify a local variable or not should only be in the function's implementation, not its interface.
- Use const *references* for types that are nontrivial and that you only need to access or pass to other const-reference functions.
- In Celeritas we use the google style of passing mutable pointers instead of mutable references, so that it's more obvious to the calling code that a value is going to be modified.
- There is rarely a reason to pass raw pointers otherwise, because (especially when other library code is in the mix) it's not always obvious what the ownership semantics are.
- CUDA `__global__` kernels change the rules *again*, you cannot have references or pointers to host memory, so usually just copy by value or provide raw pointers to host memory. To indicate that the argument should not be used inside the kernel you can prefix it with `const`, so the CUDA compiler places  the argument in `__constant__` memory rather than taking up register space.

In this case, `vol_id` is a type-safe int wrapper and so can be passed by value. Below, the `add_volume(vol_id id, GeantVolume& volume)` should be `const GeantVolume&` because the `insert` operation on the map takes a const reference and the object is large.

Reply:
Nice. Thanks for the summary.